src/lib/spinel/radio_spinel.cpp

```diff
@@ -2181,7 +2181,7 @@ void RadioSpinel::RestoreProperties(void)
         static constexpr uint16_t kFrameCounterGuard = 1000;
 
         SuccessOrDie(Set(SPINEL_PROP_RCP_MAC_FRAME_COUNTER, SPINEL_DATATYPE_UINT32_S,
-                         otLinkGetFrameCounter(mInstance) + kFrameCounterGuard));
+                         otPlatRadioGetFrameCounter(mInstance) + kFrameCounterGuard));
```

Comment:
If `otPlatxxx` isn’t suitable, I think it’s still necessary to separate `otLinkxxx` by some other method. Using a function from the OT core within the radio spinel doesn’t seem suitable. It will binds the radio spinel to the OT core, making it difficult to use the radio spinel with other protocols (like Zigbee).